taskvine/src/manager/vine_manager.c

```diff
@@ -2861,49 +2862,82 @@ static vine_result_code_t commit_task_to_worker(struct vine_manager *q, struct v
 {
```

Comment:
While I think the logic here is correct, instead of adding a loop around the existing logic, do it like this:

Create new function `commit_task_group_to_worker` that calls `commit_task_to_worker` on the desired task, and then (if needed) also calls `commit_task_to_worker` on other tasks in the same group.

Change upstream calls to `commit_task_to_worker` to `commit_task_group_to_worker` as needed.

I think that will keep the code clearer and more maintainable down the road.

Reply:
And also reduce the number of lines touched by this PR...